### ISSUE TITLE
Adjust Analytics account creation UI to cater for GA4 support

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#3253]
If the `modules/analytics-4` store is loaded (i.e. if the `ga4setup` feature flag is enabled), the Analytics account creation UI should be slightly adjusted to explain that both a UA property and a GA4 property will be created.

This ties in with #3252, but is not blocked by it. Just from a user perspective, the notification here technically only makes sense when #3252 is also implemented.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Following [this Figma design](https://www.figma.com/file/ix0pcdTQi1Bf9QFNBlBByB/GA4-%2B-UA-onboarding?node-id=20269%3A8253) (specifically pretty much only the new notification), if the `ga4setup` feature flag is enabled, the UI of the `AccountCreate` component should be adjusted:
    * The existing `GA4Notice` should be removed.
    * The new notice from the design should be displayed (using the new `SettingsNotice` props from #3251).
    * This change should not be made in `AccountCreateLegacy`, that screen can be ignored. Manual account creation as in that screen will result in a GA4 property being created anyway, and maintaining this legacy UI is not a priority other than not breaking it.

## Implementation Brief
* Using the Analytics `AccountCreate` component,
    * use the `useFeature` hook to check if the `ga4setup` feature flag is enabled.
        * If the flag is enabled, don't show the `GA4Notice` component.
        * If the flag is enabled, and once #3251 has been merged, use the `SettingsNotice` component to display the notice as per the designs in Figma using the following values for the props:
            * `icon`: The SVG of the star icon from the designs.
            * `LearnMore`: a `<Link>` component, having "Learn More" as label and URL TBD.
            * `children`: "This will create both a Google Analytics 4 and Universal Analytics property."
        * The `SettingsNotice` should be displayed after the input fields and before the "Create Account" button.

### Test Coverage
* No new tests needed.

### Visual Regression Changes
* N/A

## QA Brief

* Observe that the story in Storybook matches the design https://google.github.io/site-kit-wp/storybook/pull/3527/?path=/story/analytics-module-setup-ua-and-ga4--create-account-scope-granted

## Changelog entry

* Update Analytics account provisioning screen to inform about GA4 property creation.

**Comment by tofumatt** [comment]
> Remove the `GA4Notice` component.

I think this should only be removed when the feature flag is enabled, so I tweaked the IB to mention that, because otherwise it looks good. 👍🏻 

IB ✅

**Comment by aaemnnosttv** [comment]
## QA ⚠️ 

Overall works as expected but needs a follow up for fixing improper use of `isFeatureEnabled` due to complications with Storybook.

**Comment by wpdarren** [comment]
@aaemnnosttv when I go to [Storybook](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/analytics-module-setup-ua-and-ga4--create-account-scope-granted ) I receive a `no preview` message. 

Could you confirm what story I should be comparing to the Figma designs?  I think it might be [this one](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-analytics-setup-setupformga4--without-existing-tag)? 

![image](https://user-images.githubusercontent.com/73545194/121551064-37855980-ca07-11eb-9684-0736ca1d3263.png)

**Comment by aaemnnosttv** [comment]
@wpdarren looks like the URL to the story may have changed since it was added. The story in question is here:
https://google.github.io/site-kit-wp/storybook/develop/?path=/story/analytics-module-setup--create-account-ga4

Also worth checking other "Create Account *" stories for Analytics as well 👍

**Comment by wpdarren** [comment]
### QA Update:  ⚠️ 

@aaemnnosttv I have two observations worth noting:

1) On Storyboard we have a sentence `We’ve pre-filled the required information for your new account.` which does not appear on the Figma designs. In my opinion, this is more helpful than not including it but wanted to highlight the difference.

2) On the `Learn More` link on Storyboard we have the external link icon (see [Screenshot](https://user-images.githubusercontent.com/73545194/121696708-41b95d80-cac4-11eb-9fbe-c9b91f837723.png)) but this does not appear on the Figma designs.   

It could be for both of these observations that the Figma designs are out of date, but wanted to check. 

Thoughts?

**Comment by aaemnnosttv** [comment]
@wpdarren I believe these are expected.

> On Storyboard we have a sentence `We’ve pre-filled the required information for your new account.` which does not appear on the Figma designs. In my opinion, this is more helpful than not including it but wanted to highlight the difference.

This should probably be added to the design and may have been an oversight. I agree that it is more useful and should be kept.

> On the `Learn More` link on Storyboard we have the external link icon (see [Screenshot](https://user-images.githubusercontent.com/73545194/121696708-41b95d80-cac4-11eb-9fbe-c9b91f837723.png)) but this does not appear on the Figma designs. 

I believe this was something we decided more recently, that external links that open in a new window should (almost) always indicate this with the added icon.

@felixarntz can you confirm?

**Comment by felixarntz** [comment]
@aaemnnosttv @wpdarren Yes, both of these are correct despite the Figma files being different. The existing copy makes sense to be kept, and external links should (almost) always have that icon.

Assigning back to you @wpdarren in case there was anything else, or feel free to move to Approval if good to go otherwise.

**Comment by wpdarren** [comment]
### QA Update: Pass ✅ 

- Confirmed that the story in Storybook matches the design.
